src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -431,66 +433,40 @@ public ResultSet getCatalogs() throws SQLException {
         String dbName;
         String tableName;
         String columnName;
-        MongoJsonSchema columnSchema;
-        boolean isRequired;
+        BsonTypeInfo columnBsonTypeInfo;
+        int nullability;
         int idx;
 
         GetColumnsDocInfo(
                 String dbName,
                 String tableName,
                 String columnName,
+                MongoJsonSchema parentSchema,
                 MongoJsonSchema columnSchema,
-                boolean isRequired,
                 int idx) {
             this.dbName = dbName;
             this.tableName = tableName;
             this.columnName = columnName;
-            this.columnSchema = columnSchema;
-            this.isRequired = isRequired;
             this.idx = idx;
+
+            try {
+                this.columnBsonTypeInfo = columnSchema.getBsonTypeInfo();
+                this.nullability = parentSchema.getColumnNullability(columnName);
+            } catch (SQLException e) {
+                throw new RuntimeException(e);
```

Comment:
I will move this try-catch block to the stream code itself, instead of having it be part of this constructor.

For clarification though: do you want to reconsider the whole try-catch/`liftSQLException` pattern that is used throughout the file, or did you just want to address this particular try-catch block in this one method? If it's the former, I'll file a follow-up ticket. If it's the latter, which I'm assuming it is, I'll definitely make the change to move the try-catch to the stream code, leaving this constructor free to throw a SQLException.

Reply:
Sorry, I got fooled by the GitHub view and thought this block was in the getColumns method 🙃
Moving this try/catch block will just help with this one and is not of much value. I was looking at the broader pattern.
The try-catch/liftSQLException is our only solution for streams and it is fine. I am just concerned about missing methods which require using this pattern. 
Maybe the easiest thing we could do would be to mark methods which are throwing RuntimeException with a specific naming convention and adding a very clear comment in the method javadoc. I'll create a investigation task for this so that we can discuss and see what everyone in the team think about this.

Reply:
One idea I have is to have every overridden method in this class use `liftSQLException` and an auxiliary helper method. This would look like:
```
...

@Override
public ResultSet getColumns(...) throws SQLException {
    return liftSQLException(() -> getColumnsAux(...));
}

private ResultSet getColumnsAux(...) {
    ...
    try {
       ...
   } catch (SQLException e) {
       throw new RuntimeException(e);
   }
   ...
}

@Override
public ResultSet getTables(...) throws SQLException {
    return liftSQLException(() -> getTablesAux(...));
}

private ResultSet getTablesAux(...) {
    ...
}

...
```

Benefits of this pattern:
- we don't need to rewrite all of the methods to avoid using streams
- we can freely catch-and-wrap any SQLExceptions in any `*Aux` methods or their helpers since we'll guarantee every public method uses `liftSQLException`
- consistency across all public methods

Drawbacks of this pattern:
- it will require some documentation in code to explain the pattern
- it will still require a refactor to move the current public method bodies into private `Aux` methods and updating the public method bodies to return `liftSQLException(() -> Aux(...))`
- there may be some performance impact having every public method dispatch an anonymous function that runs in a try-catch block

Reply:
I'm sure there are also plenty of other alternatives